bin/bootstrap.py

```diff
@@ -72,52 +91,40 @@ def users(args):
-        hash_ = hashlib.sha384()
-        size = os.path.getsize(filepath)
-        try:
-            metadata = json.loads(zipfile.ZipFile(filepath).comment)
-        except ValueError as e:
-            log.warning(str(e))
-            continue
-        target, file_ = reaperutil.create_container_hierarchy(metadata)
-        with open(filepath, 'rb') as fd:
-            for chunk in iter(lambda: fd.read(2**20), ''):
-                hash_.update(chunk)
-        computed_hash = 'v0-sha384-' + hash_.hexdigest()
-        destpath = os.path.join(config.get_item('persistent', 'data_path'), util.path_from_hash(computed_hash))
-        dir_destpath = os.path.dirname(destpath)
-        filename = os.path.basename(filepath)
-        if not os.path.exists(dir_destpath):
-            os.makedirs(dir_destpath)
-        if args.copy:
-            shutil.copyfile(filepath, destpath)
-        else:
-            shutil.move(filepath, destpath)
-        created = modified = datetime.datetime.utcnow()
-        fileinfo = {
-            'name': filename,
-            'size': size,
-            'hash': computed_hash,
-            'type': 'dicom', # we are only bootstrapping dicoms at the moment
-            'created': created,
-            'modified': modified,
-            'mimetype': util.guess_mimetype(filename),
-        }
-        fileinfo.update(file_)
-        target.add_file(fileinfo)
-        rules.create_jobs(config.db, target.container, 'acquisition', fileinfo)
-
+    with requests.Session() as rs:
+        rs.verify = not args.insecure
+        rs.headers = HTTP_HEADERS
+        for dirpath, dirnames, filenames in os.walk(args.path):
+            dirnames[:] = [dn for dn in dirnames if not dn.startswith('.')] # use slice assignment to influence walk
+            if not dirnames and filenames:
+                for metadata_file in filenames:
+                    if metadata_file.lower() == 'metadata.json':
+                        filenames.remove(metadata_file)
+                        break
+                else:
+                    metadata_file = None
+                if not metadata_file:
+                    log.warning('Skipping    %s: No metadata found' % dirpath)
+                    continue
+                with open(os.path.join(dirpath, metadata_file)) as fd:
+                    try:
+                        metadata = json.load(fd)
+                    except ValueError:
+                        log.warning('Skipping    %s: Invalid metadata' % dirpath)
+                        continue
+                # FIXME need schema validation
+                with tempfile.TemporaryDirectory() as tempdir:
+                    log.info('Packaging   %s' % dirpath)
+                    filepath = create_archive(dirpath, os.path.basename(dirpath), metadata, tempdir, filenames)
+                    filename = os.path.basename(filepath)
+                    metadata.get('acquisition', {}).get('files', [{}])[0]['name'] = filename
```

Comment:
metadata could be undefined if there is no metadata.json.
this line doesn't seem also to update the metadata if it doesn't have the 'acquisition' field or the files 'subfield'.